src/game/game.cpp

```diff
@@ -10282,20 +10671,134 @@ bool Game::addItemStoreInbox(std::shared_ptr<Player> player, uint32_t itemId) {
 	}
 
 	if (internalAddItem(inboxContainer, decoKit) != RETURNVALUE_NOERROR) {
-		inboxContainer->internalAddThing(decoKit);
+		return false;
 	}
 
 	return true;
 }
 
-void Game::addPlayerUniqueLogin(std::shared_ptr<Player> player) {
-	if (!player) {
-		g_logger().error("Attempted to add null player to unique player names list");
-		return;
+bool Game::processChargesOffer(std::shared_ptr<Player> player, uint32_t itemId, uint16_t charges /* = 0*/, bool movable /* = false*/) {
+	std::shared_ptr<Item> newItem = Item::CreateItem(itemId, 1);
+	if (!newItem) {
+		return false;
 	}
 
-	const std::string &lowercase_name = asLowerCaseString(player->getName());
-	m_uniqueLoginPlayerNames[lowercase_name] = player;
+	if (charges > 0) {
+		newItem->setAttribute(ItemAttribute_t::CHARGES, charges);
+	}
+
+	if (!movable) {
+		newItem->setAttribute(ItemAttribute_t::STORE, getTimeNow());
+	}
+
+	newItem->setOwner(player);
+
+	std::shared_ptr<Thing> thing = player->getThing(CONST_SLOT_STORE_INBOX);
+	if (!thing) {
+		return false;
+	}
+
+	std::shared_ptr<Item> inboxItem = thing->getItem();
+	if (!inboxItem) {
+		return false;
+	}
+
+	std::shared_ptr<Container> inboxContainer = inboxItem->getContainer();
+	if (!inboxContainer) {
+		return false;
+	}
+
+	auto ret = internalAddItem(inboxContainer, newItem);
+	if (ret != RETURNVALUE_NOERROR) {
+		return false;
+	}
+
+	return true;
+}
+
+bool Game::processStackableOffer(std::shared_ptr<Player> player, uint32_t itemId, uint16_t amount /* = 1*/, bool movable /* = false*/) {
+	std::shared_ptr<Item> newItem = Item::CreateItem(itemId, amount);
+	if (!newItem) {
+		return false;
+	}
+
+	if (!movable) {
+		newItem->setAttribute(ItemAttribute_t::STORE, getTimeNow());
+	}
+
+	newItem->setOwner(player);
+
+	std::shared_ptr<Thing> thing = player->getThing(CONST_SLOT_STORE_INBOX);
+	if (!thing) {
+		return false;
+	}
+
+	std::shared_ptr<Item> inboxItem = thing->getItem();
+	if (!inboxItem) {
+		return false;
+	}
+
+	std::shared_ptr<Container> inboxContainer = inboxItem->getContainer();
+	if (!inboxContainer) {
+		return false;
+	}
+
+	auto ret = internalAddItem(inboxContainer, newItem);
+	if (ret != RETURNVALUE_NOERROR) {
+		return false;
+	}
+
+	return true;
+}
+
+bool Game::processNameChangeOffer(std::shared_ptr<Player> player, std::string &name) {
+	std::string newName = name;
+	trimString(newName);
+
+	auto isValidName = validateName(newName);
+
+	if (isValidName != VALID) {
+		return false;
+	}
+
+	std::ostringstream query;
+	Database &db = Database::getInstance();
+	query << "SELECT `id` FROM `player` WHERE `name` = " << db.escapeString(newName);
```

Comment:
Please prefer using fmt over ostring.
```suggestion
	Database &db = Database::getInstance();
	std::string query = fmt::format("SELECT `id` FROM `players` WHERE `name` = {}", db.escapeString(newName));
```